### DISCUSSION TITLE
MZ-Tools/FMS: worth it?

### DISCUSSION THREAD

**Discussion by hecon5** [joyfullservice/msaccess-vcs-addin#151] [Q&A] [answered]
I have seen several mentions of MZ-Tools here, and as it costs money, and I'm a cheapskate: is it worth it?

Also, FMS seems to have a ton of stuff on their site and a (quite!) expensive development basket. Are they worth it/should they be given consideration?

**Comment by Tanarri** [comment, marked as answer]
Hi @hecon5 ,

I am using MZ-Tools at the work and I think it is worth the money. There are plenty of things you could do.
Things I am using regulary is:
- ident module / project
- sort moduleprocedures
- add error handler (much variables and possibilities to create the handler like you need it)
- many code-quality options (ok, Rubberduck is here also quite good)
- code explorer
- line numbers (ok, this I do never use :-))
- create bibliothek of code
You can see here the options to image what is possible: https://www.mztools.com/v8/onlinehelp/MZTools8Help.html

Also we have purchased a copy of FMS Total Access Analyzer. Also a great tool if you want to analyze you project.
It search for a lot of errors with tables and querys and warn if you read a table/field name which doesn' exists.. and a lot more.
But it costs a lot more than MZTools.

Rubberduck is at the moment not able to analyze my access-project with 180.000 lines of code, 1100 query, 300 forms, 80 reports....
But I will always have a look to RD if some day this will be possible

**Reply by hecon5** [reply]
Thank you! This is enlightening!

**Comment by joyfullservice** [comment]
I have been using MZ-Tools on a daily basis for quite a few years, and really like it. I am using the last freeware version (v3) before they did the .Net rewrite. (v8) I tested out the beta four years ago, but it was much slower and seemed a bit cluttered at the time. I had one issue in 2016, but the developer resolved it pretty quickly. I plan to continue to using v3 as long as it works, then may switch over to the current version.

The biggest gain I see in MZ-Tools is in improving the quality of the code I write. It makes it super easy to add consistent headers and error handling to procedures (with the tap of a button) and the code review function helps me get rid of unused variables. It works great with no performance issues even on very complex projects.

I use RubberDuck occasionally, but as @Tanarri pointed out, it really bogs down on large projects. They have made a few improvements on the performance side in the past year or two, but at this point it slows down the VBA IDE too much for me to use it on a regular basis.

Another interesting tool that I have found helpful at times is [V-Tools](https://www.skrol29.com/us/vtools.php). In particular, it can be helpful for reviewing and editing import/export specifications.

**Reply by hecon5** [reply]
Most excellent!
Thank you!

**Comment by A9G-Data-Droid** [comment]
The premier error handler is: [vbWatchDog](https://www.everythingaccess.com/vbwatchdog.asp)
I daily drive [RubberDuckVBA ](https://rubberduckvba.com/)for years now.

I've never seen a reason to use MZ-Tools.
